draft-ietf-privacypass-protocol.md

```diff
@@ -34,962 +34,566 @@ author:
+    uint8_t nonce[32];
+    uint8_t context[32];
+    uint8_t key_id[32];
+    uint8_t authenticator[Nk];
+} Token;
 ~~~
 
-Note that a `IssuanceInput` contains equal-length size `m` arrays of
-`Internal` and `IssuanceRequest` types corresponding to the number of
-tokens that should be issued. `data_bytes` is the total number of bytes
-that `m` `Internal` elements require, and `req_bytes` is the total
-number of bytes that `m` `IssuanceRequest` elements require.
+Otherwise, the Client aborts the protocol.
 
-### IssuanceResponse {#pp-srv-issue-response}
+## Issuer Configuration
 
-Firstly, the `IssuedToken` type corresponds to a single sequence of
-bytes that represents a single issued token received from the server.
+Issuers are configured with Private and Public Key pairs, each denoted skI and
+pkI, respectively, used to produce tokens. Each key pair MUST be generated as
+follows:
 
 ~~~
-opaque IssuedToken<1..2^16-1>
+(skI, pkI) = GenerateKeyPair()
 ~~~
 
-Then an `IssuanceResponse` corresponds to a collection of `IssuedTokens`
-as well as a sequence of bytes `proof`.
+The key identifier for this specific key pair, denoted `key_id`, is computed
+as follows:
 
 ~~~
-struct {
-  IssuedToken tokens[tokens_bytes]
-  opaque proof<1..2^16-1>
-}
+key_id = SHA256(0x0004 || SerializeElement(pkI))
 ~~~
 
-The value of `m` is equal to the length of the `IssuanceRequest` vector
-sent by the client. `tokens_bytes` is the total number of bytes that `m`
-`IssuedTokens` elements require.
+# Issuance Protocol for Publicly Verifiable Tokens {#public-flow}
 
-### RedemptionToken {#pp-redemption-token}
+This section describes a variant of the issuance protocol in {{private-flow}}
+for producing publicly verifiable tokens. It differs from the previous variant
+in two important ways:
 
-The `RedemptionToken` struct contains the data required to generate the
-client message in the redemption phase of the Privacy Pass protocol.
+1. The output tokens are publicly verifiable by anyone with the Issuer public
+   key; and
+1. The issuance protocol does not admit public or private metadata to bind
+   additional context to tokens.
 
-~~~
-struct {
-  opaque data<1..2^16-1>;
-  opaque issued<1..2^16-1>;
-} RedemptionToken;
-~~~
+Otherwise, this variant is nearly identical. In particular, Issuers provide a
+Private and Public Key, denoted skI and pkI, respectively, used to produce tokens
+as input to the protocol. See {{public-issuer-configuration}} for how this key
+pair is generated.
 
-### RedemptionRequest {#pp-redemption-message}
+Clients provide the following as input to the issuance protocol:
 
-The `RedemptionRequest` struct consists of the data that is sent by the
-client during the redemption phase of the protocol.
+- Issuer name, identifying the Issuer. This is typically a host name that
+  can be used to construct HTTP requests to the Issuer.
+- Issuer Public Key pkI, with a key identifier `key_id` computed as
+  described in {{issuer-configuration}}.
```

Comment:
This reference is wrong, since it's pointing to the VOPRF text.

Reply:
Fixed 👍